backend/migrations/000002_checkpoints_update_schema.up.sql

```diff
@@ -1,11 +1,41 @@
-ALTER TABLE checkpoints
-    ADD COLUMN IF NOT EXISTS thread_ts TIMESTAMPTZ,
-    ADD COLUMN IF NOT EXISTS parent_ts TIMESTAMPTZ;
+-- Drop existing checkpoints-related tables if they exist
```

Comment:
I believe this migration fails for anyone that's run migrations 1 through 4 already. the migration state is kept of in the database. 

Should this run as step 5 so it'll run at the end?

So if you try to run opengpts with the previous version, and then apply PR on top and run things -- the migrations will not work.

---

W/ current approach it seems like any old threads are no longer usable from the app. (I'm assuming not super easy to recover b/c of the pickle serde that was used).

Reply:
You’re right! I think the same thing happens in LangGraph when people decide to use the new checkpointer, right?

Reply:
The new checkpointers can be versioned as far as I understand

https://github.com/langchain-ai/langgraph/blob/main/libs/checkpoint-postgres/langgraph/checkpoint/postgres/base.py#L27

So at least going forward there's a way to carry out schema migrations automatically.

But yeah going from the pickle checkpointer -> new checkpointer was a breaking change. I'm OK if we don't worry about this, don't think this affects that many users.

I'd just prefer if we didn't wipe out any potential sql tables that users may want to recover data from